packages/jellyfish-api-core/__tests__/category/wallet/getBalances.test.ts

```diff
@@ -0,0 +1,78 @@
+import { MasterNodeRegTestContainer, RegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { BigNumber } from '../../../src'
+import { WalletFlag } from '../../../src/category/wallet'
+
+describe('getBalances on masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should getBalances', async () => {
+    const balances = await client.wallet.getBalances()
+    expect(balances.mine.trusted).toBeInstanceOf(BigNumber)
+    expect(balances.mine.untrusted_pending).toBeInstanceOf(BigNumber)
+    expect(balances.mine.immature).toBeInstanceOf(BigNumber)
+    expect(typeof balances.mine.used).toStrictEqual('undefined')
+
+    expect(typeof balances.watchonly).toStrictEqual('undefined')
+  })
+
+  it('should have used in getBalances when wallet is set to avoid_reuse', async () => {
+    await client.wallet.setWalletFlag(WalletFlag.AVOID_REUSE)
+    await container.generate(1)
+    const balances = await client.wallet.getBalances()
+
+    expect(balances.mine.trusted).toBeInstanceOf(BigNumber)
+    expect(balances.mine.untrusted_pending).toBeInstanceOf(BigNumber)
+    expect(balances.mine.immature).toBeInstanceOf(BigNumber)
+    expect(balances.mine.used).toBeInstanceOf(BigNumber)
+
+    expect(typeof balances.watchonly).toStrictEqual('undefined')
+  })
+
+  it('should show balances after transfer', async () => {
+    const balance = await client.wallet.getBalances()
+
+    const address = await client.wallet.getNewAddress()
```

Comment:
Any address that is generated by `client.wallet.getNewAddress()` will still belong to you address pool hence be under your balance.

If you want to see balance away, aka balance you can't spend you need to send it to an address that the node can't solve.

Here is a list of address you can use: https://gist.github.com/fuxingloh/a0973c5529a70a47cbe65ec02aaab600

Reply:
For balance going in, you can move balance out then move it back in.

This is how you can do it.

https://github.com/DeFiCh/jellyfish/blob/main/packages/jellyfish-api-core/__tests__/category/rawtx/sendRawTransaction.test.ts